osgar/launcher.py

```diff
@@ -35,13 +33,6 @@ def launch(app, description, prefix):
         game.run()
 
     elif args.command == 'run':
-        log = LogWriter(prefix=prefix, note=str(sys.argv))
-        config = config_load(*args.config)
-        log.write(0, bytes(str(config), 'ascii'))  # write configuration
-        robot = Recorder(config=config['robot'], logger=log, application=app)
-        game = robot.modules['app']  # TODO nicer reference
-        robot.start()
-        game.run()
-        robot.finish()
+        record(args.config, log_prefix=prefix, note=str(sys.argv), application=app)
```

Comment:
Should the whole command line be stored as a note? I would expect `arg.note` there instead of `str(sys.argv)`. If this is so, then also the `import sys` statement is moot.

Reply:
Hmm, why is it still different from `replay`? Is there any advantage to having the `run()` call inside in `record` and outside in `replay`?

Reply:
Regarding `--note` ... more I think about it I would just hide it inside `record()` function. It is more useful to have whole command line (i.e. what module was running, what config files were selected) + if there is `--note` you will see it there too.

`replay` is very different from `record`. While `replay` is only on module or application, record is always all modules. For debugging of single module it is easier to run everything in single thread, i.e. skip `start()` and directly call `run()`.

Reply:
note removed in fixup

Reply:
Well, here I actually meant that we should maybe return `game` out from `record` and call `game.run` outside, as we do for replay - so that the user is "in charge" (and not don't call us, we will cal you). It is also tangentially related to the problem we are debugging right now, where it appears, that if the the `application` is inheriting from `Thread`, it runs twice.